src/slptokens/token.cpp

```diff
@@ -0,0 +1,472 @@
+// Copyright (c) 2019 Greg Griffith
+// Copyright (c) 2019 The Bitcoin Unlimited developer
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "token.h"
+#include "cash_protocols.h"
+
+bool CSLPToken::DetermineDynamicSize(const CScript &scriptPubKeyIn, size_t &index, uint32_t &return_size)
+{
+    return_size = 0;
+    uint8_t size = scriptPubKeyIn[index];
+    index++;
+    // no empty-push opcode or 1-byte literal push opcodes allowed
+    if (size == 0x00 || (size >= 0x4f && size <= 0x60))
+    {
+        return_size = 2;
+        return false;
+    }
+    if (size == 0x4c) // OP_PUSHDATA1 - 0x4c 0x04 [chunk]
```

Comment:
why not use the symbolic names here?

Reply:
I found that putting in the hex values helped a lot during debugs when i had print statements added since i was printing out the hex values of numbers. the SLP spec also has a lot of these values in hex format and not using the opcode names in their examples so it was easier to compare there. 

I don't have the op codes memorized so if i were to come back and look at this later i would have to go translate them to the hex values in the script headers anyway. i just thought this was easier